### DISCUSSION TITLE
8 Relay Outputs

### DISCUSSION THREAD

**Discussion by lexa7741** [forkineye/ESPixelStick#350] [General]
How to use it? There are no 8 relays in the settings.

**Comment by MartinMueller2003** [comment]
The last output port in the device config is used to enable support for the external relay port device. You connect a PCA9685 to the I2C bus.
    On Sunday, September 12, 2021, 04:06:10 PM EDT, lexa7741 ***@***.***> wrote:  
 
 


How to use it? There are no 8 relays in the settings.

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Reply by lexa7741** [reply]
> The last output port in the device config is used to enable support for the external relay port device. You connect a PCA9685 to the I2C bus. 

I don't need 16 PWM Outputs (I2C bus).
I need 8 Relay Outputs.

**Comment by MartinMueller2003** [comment]
A relay output is done by sending a PWM output 0 for off and 255 for on. Alternatively you can select the Relay option and set the GPIOs that you want to use. That will use up to 8 of the internal GPIOs. The trigger thresh hold is used to determine on/off. Over trigger is on, under trigger is off.
Martin
    On Sunday, September 12, 2021, 04:57:43 PM EDT, lexa7741 ***@***.***> wrote:  
 
 



The last output port in the device config is used to enable support for the external relay port device. You connect a PCA9685 to the I2C bus.


I don't need 16 PWM Outputs (I2C bus).
I need 8 Relay Outputs.

—
You are receiving this because you commented.

Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Reply by lexa7741** [reply]
> A relay output is done by sending a PWM output 0 for off and 255 for on. Alternatively you can select the Relay option and set the GPIOs that you want to use. That will use up to 8 of the internal GPIOs. The trigger thresh hold is used to determine on/off. Over trigger is on, under trigger is off. 

I thought there was a simple solution like in this author's project https://github.com/garymueller/ESP-131GPIO

**Comment by brmeadows2841** [comment]
After the output is set for the GIPOS on the "Output 8 - Relay Configuration page" how is the GPIOs turned on/off. Is there a control page somewhere in the Espixelstick firmware?

**Reply by MartinMueller2003** [reply]
For ESP32 builds, the last port shows relay support. You can select it and then set the GPIOs you want to use.

**Reply by MtnMan66** [reply]
I've figured out how to configure the ESP32 for relay support, but don't see how to configure xLights to take advantage of this.  It looks like xLights only allows 6 output ports for espixelstick ESP32.  Is there some workaround in xLights to make this work?

**Reply by MartinMueller2003** [reply]
Configure the entire relay port as a single manually configured port with N channels of AC dimable lights. Each relay output is just like an AC string. Then configure the Relay channel with w trip level of 128. Any value above that = on. all others = off.
